integration_test/ops_agent_test.go

```diff
@@ -2475,14 +2485,13 @@ func runResourceDetectorCli(ctx context.Context, logger *logging.DirectoryLogger
 	}
 
 	// Run the resource detector in the VM
-	goPath, err := InstallGolang(ctx, logger, vm)
-	if err != nil {
+	if err := installGolang(ctx, logger, vm); err != nil {
 		return nil, err
 	}
 	cmd := fmt.Sprintf(`
 		%s
 		cd %s
-		go run run_resource_detector.go`, goPath, workDir)
+		go run run_resource_detector.go`, goPathCommandForPlatform(vm.Platform), workDir)
```

Comment:
It seems possible to refactor the code here to use the new `runGoCode` to test the `resourcedetector`, but probably it is too much for this PR. I could be done in a subsequent PR.

Reply:
Yeah, I have some other ideas too for simplifying that test which could avoid needing to upload Go code at all for it -- we can definitely do that in a later PR.